deploy/poseidon-deployment.yaml

```diff
@@ -1,21 +1,130 @@
-apiVersion: v1
-kind: ServiceAccount
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
 metadata:
+  labels:
+    kubernetes.io/bootstrapping: rbac-defaults
+  name: system:poseidon
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: system:poseidon
+subjects:
+- kind: ServiceAccount
   name: poseidon
   namespace: kube-system
 ---
-kind: Service
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    kubernetes.io/bootstrapping: rbac-defaults
+  name: system:poseidon
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - events
+  verbs:
+  - create
+  - patch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - endpoints
+  verbs:
+  - create
+- apiGroups:
+  - ""
+  resourceNames:
+  - poseidon
+  resources:
+  - endpoints
+  verbs:
+  - delete
+  - get
+  - patch
+  - update
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  verbs:
+  - delete
```

Comment:
Does scheduler need to delete Pod?

Reply:
Yes, Poseidon uses delete to evict a running pod.
This was an experimental thing which we tried for preemption based on some priority.
We will use pod evict function in the future, but for now, we need this.